### PR TITLE
Only create more append_vecs when the account number grows

### LINKED ISSUE #5432: Out of memory when mapping new accounts append vec
#### Problem

```
[2019-08-06T15:51:48.892717417Z INFO  solana::replay_stage] new fork:6494 parent:6427
thread 'solana-replay-stage' panicked at 'failed to map the data file: Os { code: 12, kind: Other, message: "Cannot allocate memory" }', src/libcore/result.rs:999:5
```

Overall memory usage is low:
```
              total        used        free      shared  buff/cache   available
Mem:           187G        1.9G        182G         41M        3.0G        184G
```

#### Proposed Solution

Believe it might be vm_max_map_count being hit.

One workaround could be increasing the map count which can help the issue:
`sysctl -w vm.max_map_count=262144`

Otherwise append_vecs may need to be unmapped or cleaned more aggressively when they are not used.

### PR DESCRIPTION
**Pull request by sakridge**
### Problem

Linux doesn't like when we mmap too much.
We only need many append_vecs if the number of accounts is high,
so only create opportunistic ones as accounts are created.

#### Summary of Changes

Only create opportunistic ones as accounts are created.

Fixes #5432

### CODE REVIEW

File: runtime/src/accounts_db.rs [hunk 1]
```diff
@@ -586,16 +586,34 @@ impl AccountsDB {
...
         if let Some(fork_stores) = stores.0.get(&fork_id) {
             if !fork_stores.is_empty() {
+                if fork_stores.len() <= self.min_num_stores {
+                    let mut total_accounts = 0;
+                    for store in fork_stores.values() {
+                        total_accounts += store.count_and_status.read().unwrap().0;
+                    }
+
+                    // Create more stores so that when scanning the storage all CPUs have work
+                    if (total_accounts / 16) >= fork_stores.len() {
```
Comment (rob-solana) [inline runtime/src/accounts_db.rs]: `>=` ?  not sure how it matters much, 16 is kind of arbitrary, might we prefer a comparison with min_num_stores instead?

**Comment by mvines** [comment]
(FWIW, I just observed #5432 on the edge testnet)

File: runtime/src/accounts_db.rs (hunk 1 above)
Comment (sakridge) [reply runtime/src/accounts_db.rs]: Yea, the >= is so once we hit 16 accounts total, then we create a new store. Another at 32, 48, 64, etc.

Maybe something like `2048 / (32 + min_num_stores)`

Comment (sakridge) [reply runtime/src/accounts_db.rs]: That would give something like this:

min_num_stores | accounts_rate
-- | --
2 | 60
4 | 57
8 | 51
16 | 43
32 | 32
64 | 21

Comment (sakridge) [reply runtime/src/accounts_db.rs]: Those numbers are largely arbitrary and probably doesn't make much difference. Can go with just 16 for now.

**Pull request merged**